datastore/save.go

```diff
@@ -138,6 +138,17 @@ func reflectFieldSave(props *[]Property, p Property, name string, opts saveOpts,
 		}
 	}
 
+	if v.CanAddr() {
+		vi := v.Addr().Interface()
+		if pSaver, ok := vi.(PropertyLoadSaver); ok {
+			val, err := pSaver.Save()
+			if err != nil {
+				return fmt.Errorf("datastore: unsupported struct field type: %v", v.Type())
+			}
+			p.Value = val
+		}
+	}
+
```

Comment:
I've tried to use the existing functions (which are doing the similar things), but they doesn't seem to be correct solution for this case